### PR TITLE
upgrade the alpine base image

### PR DESCRIPTION
**Pull request by yanggangtony**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Current Alpine Version 3.18.2 (Released Jun 14, 2023)
[alpinelinux downloads](https://alpinelinux.org/downloads/).
And the dockerhub image is 3.18 and 3.18.0.
So we can follow the latest 3.18.
[dockerhub](https://hub.docker.com/_/alpine/tags)

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
The base image `alpine` now has been promoted from `alpine:3.17.1` to `alpine:3.18.2`.
```

### CODE REVIEW

**Comment by yanggangtony** [comment]
```
[2023-06-14T16:12:02.263Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
[2023-06-14T16:12:02.264Z] ['verbose'] The error stack is: Error: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
    at main (/snapshot/repo/dist/src/index.js)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2023-06-14T16:12:02.264Z] ['verbose'] End of uploader: 1052 milliseconds
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255
```



the ci errors..
is the network interrelated?

File: cluster/images/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.17.1
+FROM alpine:3.18
```
Comment (RainbowMango) [inline cluster/images/Dockerfile:1]: I'd prefer the 3.18.2 which is available now:
https://hub.docker.com/layers/library/alpine/3.18.2/images/sha256-25fad2a32ad1f6f510e528448ae1ec69a28ef81916a004d3629874104f8a7f70?context=explore

Comment (yanggangtony) [reply cluster/images/Dockerfile:1]: good

Comment (yanggangtony) [reply cluster/images/Dockerfile]: Rebased.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
By the way,

> the ci errors..
is the network interrelated?

Sort of. You can refer to https://github.com/karmada-io/karmada/blob/4e4bfffc61067ddbde6b287cdb338bea2f5647df/.github/workflows/ci.yml#L98-L100 for more details.
It's kind of an annoying thing that nothing we can do about it.

I thinking if we should set the `fail_ci_if_error` with `false`, because it doesn't matters.

**Comment by yanggangtony** [comment]
Thanks for explain.

**Pull request merged**